src/metadata.ts

```diff
@@ -0,0 +1,34 @@
+import * as constants from './lib/constants';
+import * as request from './lib/request';
+import { readFileSync, writeFileSync } from 'fs';
```

Comment:
Don't ever use `Sync` functions in the supervisor, use the promise based ones:

```suggestion
import { promises as fs } from 'fs';
```

and then 
`fs.readFile`, fs.writeFile`